packages/docs/community/release-process/smart-contracts.md

```diff
@@ -119,18 +131,21 @@ celocli governance:show --proposalID <proposalId> --jsonTransactions "upgrade_pr
 
 ### Verify Proposed Release
 
-Verify that the proposed upgrade activates contract addresses which match compiled bytecode from the tagged release candidate exactly. Include initialization data from the CGP to verify via (add `-i initialization_data.json` in that case)
+This script serves the same purpose as `verify-deployed` but for a not-yet
+accepted contract upgrade (in the form of the proposal.json you fetched in the step prior). It gives you the confidence that the branch specified in the `-b` flag in (same as `check-versions`) will be the resulting network state of the proposal if executed. It does so by going over all Celo Core Contracts and determining updates to the Registry pointers, proxy or implementation contracts and verifying their implied bytecode against the compiled source code.
+
+Additionally, include `initialization_data.json` from the CGP if any of the contracts have to be initialized.
 
 ```bash
 RELEASE_CANDIDATE="celo-core-contracts-v${N}.rc${X}"
 NETWORK=${"baklava"|"alfajores"|"mainnet"}
 # A -f boolean flag can be provided to use a forno full node to connect to the provided network
-yarn verify-release -p "upgrade_proposal.json" -b $RELEASE_CANDIDATE -n $NETWORK -f
+yarn verify-release -p "upgrade_proposal.json" -b $RELEASE_CANDIDATE -n $NETWORK -f -i initialization_data.json
```

Comment:
is `initialization_data.json` referring to the same file as `releaseData/initializationData/release${N}.json` on 105?  I referred to line 105 in an above comment too. Seems to be the only area of confusion for me as I read this!

Reply:
This should be a different file and I also struggled with specifying that appropriate. It's again because the deployer and the verifiers run the scripts basically the same, but not exactly the same. In this case, verifiers should fetch the initialization data from the CGP itself.